### PR TITLE
[MRG] Fix quadruplets scoring

### PR DESCRIPTION
**Pull request by wdevazelhes**
There was a problem with scoring in LSML: this PR fixes it (and adds a non-regression test as well as a small test for prediction for pairs learners)

### CODE REVIEW

File: metric_learn/base_metric.py [hunk 1]
```diff
@@ -641,4 +641,4 @@ def score(self, quadruplets):
     score : float
       The quadruplets score.
     """
-    return - np.mean(self.predict(quadruplets))
+    return np.mean((self.predict(quadruplets) + 1) / 2)
```
Comment (perimosocordiae) [inline metric_learn/base_metric.py]: This should be equivalent to `pred.mean() / 2  + 0.5`, which is more efficient to compute.

I'd also add a comment explaining that the prediction is a vector of either -1 or 1 values, and that we're rescaling to [0, 1].

Comment (wdevazelhes) [reply metric_learn/base_metric.py]: I agree, done

**Comment by wdevazelhes** [comment]
If everybody agrees, I think we are good to merge

**Comment by perimosocordiae** [comment]
Agreed, merged.

**Pull request merged**